docs/src/content/docs/basic-concepts.mdx

```diff
@@ -0,0 +1,116 @@
+---
+title: Basic Concepts
+description: Understanding the core concepts of GenSX
+---
+
+GenSX is a workflow engine designed for building complex LLM applications. It's built around functional, reusable components that are composed using JSX to create and orchestrate complicated workflows.
+
+## Components
+
+Components are the building blocks of GenSX applications; they're pure TypeScript functions that:
+
+- Accept props as input
+- Return a single output or another component
```

Comment:
I might say: 

"Produces an output that can be used by another component"
Perhaps separately or in parenthesis add "(and outputs can be composed of other components!)"